flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/KinesisConfigUtil.java

```diff
@@ -244,7 +244,11 @@ public static void validateAwsConfiguration(Properties config) {
 		}
 
 		if (!config.containsKey(AWSConfigConstants.AWS_REGION)) {
-			throw new IllegalArgumentException("The AWS region ('" + AWSConfigConstants.AWS_REGION + "') must be set in the config.");
+			if (!config.containsKey(ConsumerConfigConstants.AWS_ENDPOINT)) {
+				// per validation in AwsClientBuilder
+				throw new IllegalArgumentException(String.format("Either AWS region ('%s') or AWS endpoint ('%s') must be set in the config.",
+					AWSConfigConstants.AWS_REGION, AWSConfigConstants.AWS_REGION));
+			}
 		} else {
```

Comment:
Do we also need to check that not both `AWS_REGION` and `AWS_ENDPOINT` is set?
(Since the AwsClientBuilder says that ONLY ONE of these 2 may be set).